basket/news/utils.py

```diff
@@ -95,14 +100,66 @@ def email_is_blocked(email):
     return False
 
 
+def is_authorized(request, email=None):
+    if has_valid_api_key(request):
+        return True
+
+    if email and has_valid_fxa_oauth(request, email):
+        return True
+
+    return False
+
+
 def has_valid_api_key(request):
     # The API key could be the query parameter 'api-key' or the
     # request header 'X-api-key'.
-
     api_key = (request.REQUEST.get('api-key', None) or
                request.REQUEST.get('api_key', None) or
                request.META.get('HTTP_X_API_KEY', None))
-    return APIUser.is_valid(api_key)
+    if api_key:
+        return APIUser.is_valid(api_key)
+
+    return False
+
+
+def has_valid_fxa_oauth(request, email):
+    if not email:
+        return False
+
+    server_urls = fxa.constants.ENVIRONMENT_URLS.get(settings.FXA_OAUTH_SERVER_ENV)
+    if not server_urls:
+        return False
+
+    # Grab the token out of the Authorization header
+    authorization = request.META.get('HTTP_AUTHORIZATION')
+    if not authorization:
+        return False
+
+    authorization = authorization.split(None, 1)
+    if authorization[0].lower() != 'bearer' or len(authorization) != 2:
+        return False
+
+    token = authorization[1]
+    oauth = fxa.oauth.Client(server_url=server_urls['oauth'])
```

Comment:
By default, this `oauth.Client` instance will keep a little in-memory cache of recently-seen tokens.  I'm not sure that'll be worth it for your use-case here, where you don't really expect to see multiple requests with the same token.  If you like can disable it by passing `cache=None` in the constructor here.

Reply:
I've looked at the code and the cache has a default TTL of 5 min, so I think it seems like no harm to keep and possibly useful in some situations. Leaving the defaults. Thanks for the tip!

Reply:
Right, it expires aggressively, but there's no upper limit on the number of items in the cache if you happen to have many millions of unique accesses in that 5-minute window.

Reply:
Hopefully you're not planning on that ;)